packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
+  render(<FormGroup>{<CounterMock value={5} error={CustomError} />}</FormGroup>);
 
-  expect(queryByTestId('test')).toBeInTheDocument();
+  expect(screen.getByRole('link')).toBeInTheDocument();
 });
 
 test('does not accept non-Error Components', () => {
   const NotAnError = (
     <div>
       This is a not a custom error component
-      <a href="#" data-testid="test">
-        has a url
-      </a>
+      <a href="#">has a url</a>
     </div>
   );
 
-  const { queryByTestId } = render(<FormGroup>{counterMock({ error: NotAnError, ...requiredAttributes })}</FormGroup>);
+  render(<FormGroup>{<CounterMock value={5} error={NotAnError} />}</FormGroup>);
 
-  expect(queryByTestId('test')).not.toBeInTheDocument();
+  expect(screen.queryByRole('link')).not.toBeInTheDocument();
 });
 
 test('renders both the add and subtract icons', () => {
-  const { getAllByRole } = render(counterMock(requiredAttributes));
+  render(<CounterMock value={5} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons.length).toBe(2);
+  expect(buttons).toHaveLength(2);
 });
 
 test('buttons are disabled when value hits max or min', () => {
-  const { getAllByRole, rerender } = render(counterMock({ ...requiredAttributes, value: 0 }));
+  const { rerender } = render(<CounterMock value={0} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons[0]).toHaveProperty('disabled', true);
-  expect(buttons[1]).toHaveProperty('disabled', false);
+  expect(buttons[0]).toBeDisabled;
+  expect(buttons[1]).toBeEnabled;
 
-  rerender(counterMock({ ...requiredAttributes, value: 10 }));
+  rerender(<CounterMock value={10} />);
 
-  expect(buttons[0]).toHaveProperty('disabled', false);
-  expect(buttons[1]).toHaveProperty('disabled', true);
+  expect(buttons[0]).toBeEnabled;
+  expect(buttons[1]).toBeDisabled;
 });
 
 test('value prop only accepts whole numbers', () => {
-  const { container } = render(counterMock({ max: 20, ...requiredAttributes }));
+  render(<CounterMock value={5} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
+  const fractionalValue = '1.7';
 
-  const input = container.getElementsByTagName('input');
-  fireEvent.change(input[0], { target: { value: 1.5 } });
+  userEvent.clear(input);
+  userEvent.type(input, fractionalValue);
 
-  expect(handleChange).toHaveBeenCalledWith(2);
+  expect(input).toHaveValue('17');
 });
 
 test('value increases when increase or decrease icons are clicked', () => {
-  const { getByDisplayValue, container } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5') as HTMLInputElement;
-
-  const icons = container.getElementsByTagName('svg');
+  const counterValue = screen.getByRole<HTMLInputElement>('textbox').value;
```

Comment:
Instead of specifying `<HTMLInputElement>`, grab the whole element and expect with `toHaveValue()`

```tsx
const counter = screen.getByRole<HTMLInputElement>('textbox');

expect(counter).toHaveValue('5');
```